### DISCUSSION TITLE
13 June 2024 1400

### DISCUSSION THREAD

**Discussion by ktoddbrown** [ktoddbrown/SoilDRaH#19] [RegularCalls]
# (5 | 1400) Check ins

 - Who is here
 - Note taking?
 - How are you showing up? [Feels wheel](https://feelingswheel.com/)
 
# (45 | 1405) Issues checkins and stuck to unstuck

Pull up open issues and walk through in reverse order.

# (5 | 1450) Soil-DRaH Wiki

# (5 | 1455) Check outs [appreciations, curiosities, and ToDo]

**Comment by vaasukimarupaka** [comment]
### (5 | 1400) Check ins

- Who is here: Gabriel, Kathe,  Kate, Vaasuki
- Note taking? Vaasuki
- How are you showing up? [Feels wheel](https://feelingswheel.com/)
### (45 | 1405) Issues checkins and stuck to unstuck
Pull up open issues and walk through in reverse order.

1. Vaasuki: Updates on CPEAT
Finished commenting the updated readCPEAT function 
The current function does not load in the of_variable column into the final data table; so merge that in to the function
Currently working on the CPEAT report and generating figures for the datasets
Ordering of the data table will be different for different datasets, so it is not so worrisome!

1. KTB: working on ISCN3 function
read function up and running
figures are ready
Gabriel to follow up by commenting the codebase; refer to Report_ISCN3.Rmd file under vignettes folder
Also work up the comments for the readISCN3.R function
KTB going to work with check annotations issue 

1. Gabriel:
Will work on commenting on the codebase (assigned on GitHub)
Working on the Longleaf Pune biomass/carbon data 
Geochemical and mineralogical data; soil A horizon (CONUS)
Annotations_saved.df
Write a function that only works for the specific dataset (as the data structure is so different for different datasets)
Look up ’Stringr' function in r to work with regular expressions 
Put some of the r script and put it in the report (a new file maybe?) and have a chunk that says: Construct annotations 
Focus on hard coding some of these parses
Split the list on the delimiter ‘;’ 
Work for matches between the headers and the first word in the semicolon
Try to avoid hard coding indices 
Create a temp directory and work there?
The curated version will go into the data folder
First 10 rows look like control vocabulary; so ignore them 

1. Kate:
Will read through and review the read scripts 
ISRaD annotations:
Learning to code a bit in R for this database!!
Starting with layer table
Start an ISRaD report and at the bottom put the R code that you are building in this report
Push this to a new branch [KTB creating a branch now! - 17-ISRaD2020}
Do not commit data files to the repo (css, xml etc) until we got our annotations finalized 
Shall walk thru extending this the other tables in the DB going forward
### (5 | 1450) Soil-DRaH Wiki
Any questions or tension points, please feel free to drop those comments/suggestions on the wiki page!
### (5 | 1455) Check outs [appreciations, curiosities, and ToDo]
For loops are really inefficient for R!!
